pkgs/os-specific/linux/nvidia-x11/generic.nix

```
@@ -1,8 +1,11 @@
 { version
+, url ? null
 , sha256_32bit ? null
 , sha256_64bit
 , settingsSha256
+, settingsVersion ? null
```

```suggestion
, settingsVersion ? version
```
would remove the need for `...}@args` and `persistencedVersion = args.persistencedVersion or version;`

---

good idea! done (args is still required for `args.url or ...`)

---

Actually, you could compare `url` with `null`. IIRC the use of `{ }@args` with `callPackage` is discouraged, I don’t remember the reason of the top of my head.

---

On a second thought, `@args` should be fine as long as we don’t have an ellipsis `, ... }@args`